File: stress/t/arr_reverse.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,j,n,a,b,s,t;
  int64_t A[20];
  n=20;
  for(i=0;i<n;i++){ A[i]=i*i+1; }
  i=0; j=n-1;
  while(i<j){ a=A[i]; b=A[j]; A[i]=b; A[j]=a; i=i+1; j=j-1; }
  s=0;
  for(i=0;i<n;i++){ t=A[i]; s=s+t*(i+1); }
  printf("wsum="); printf("%ld",(long)s); printf("\n");
  t=A[0]; printf("first="); printf("%ld",(long)t); printf("\n");
  t=A[19]; printf("last="); printf("%ld",(long)t); printf("\n");
  return 0;
}

File: stress/t/arr_reverse.f
      PROGRAM ARRREVERSE
      INTEGER*8 I,J,N,A,B,S,T
      INTEGER*8 AR(0:19)
      N=20
      I=0
   10 IF (I.GE.N) GOTO 20
      AR(I)=I*I+1
      I=I+1
      GOTO 10
   20 CONTINUE
      I=0
      J=N-1
   30 IF (I.GE.J) GOTO 40
      A=AR(I)
      B=AR(J)
      AR(I)=B
      AR(J)=A
      I=I+1
      J=J-1
      GOTO 30
   40 CONTINUE
      S=0
      I=0
   50 IF (I.GE.N) GOTO 60
      T=AR(I)
      S=S+T*(I+1)
      I=I+1
      GOTO 50
   60 CONTINUE
      WRITE(*,'(A,I0)') 'wsum=',S
      T=AR(0)
      WRITE(*,'(A,I0)') 'first=',T
      T=AR(19)
      WRITE(*,'(A,I0)') 'last=',T
      END
